src/savepng.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+ * SDL_SavePNG -- libpng-based SDL_Surface writer.
```

Comment:
You could add here and in header:
Code taken from driedfruit: https://github.com/driedfruit/SDL_SavePNG